### ISSUE TITLE
train-test in mvnn

### ISSUE THREAD

**Issue by eeyhsong** [gifale95/eeg_encoding#2]
Hi, Alessandro,

Really appreciate that your group provided such a wonderful dataset and detailed processing code!

There is a little issue at `01_eeg_preprocessing/preprocessing_utils.py Line 167`. Here, it uses both training and test data for the whiten matrix. I think it will be better to use only the training part for the matrix for some tasks. May replace with `sigma_tot = sigma_part[1]`. It has no obvious impact on the results of my test.

BTW, we've finished a self-supervised framework for multi-class decoding with your dataset, https://arxiv.org/abs/2308.13234. Hope to talk with you about the visual systems and computational models someday. 

Best wishes,
Song

**Comment by gifale95** [comment]
Hi Song,

Cool that you've been using our dataset!

And thank you for pointing out the MVNN issue. It was actually intentional to do it as we did. Although I agree that in some scenarios one might want to completely prevent any data leakage between train and test partitions, I do not think that in our case this was problematic. This is because, before being applied, the whitening matrix is averaged over the EEG responses for all exemplar images (both train and test), and as a consequence contains noise-covariance information common across images. On the other hand, our modeling analyses (encoding and decoding) are specifically testing idiosyncrasies of EEG responses for individual exemplar images. In sum, given our analyses, I don't think computing the whitening matrices the way I did is problematic since it did not lead to leakage of information related to individual exemplar images (if not also because the train and test partition of the dataset not only don't share the same exemplars withing an object concept, but don't even share the same object concepts). But again, this all depends on the context, and there likely are situations where the approach we applied could be problematic.

I will attend the VSS and CCN conferences this year, perhaps I will see you there!
